src/algorithms/graph-builder.ts

```diff
@@ -0,0 +1,37 @@
+import {INode, Node} from '../api/schema/node';
+import {IWay, Way} from '../api/schema/way';
+
+function randNode(id: number) {
+  return new Node({
+    _id : id,
+    loc : {
+      type : 'Point',
+      coordinates : [Math.random() * 10 + 1, Math.random() * 60 + 1]
+    }});
+}
+
+function randWay(id: number, wayNodes: INode[]) {
+  return new Way({
+    _id : id,
+    tags : {
+      highway : 'primary'
+    },
+    loc : {
+      type : 'Polygon',
+      coordinates : [],
+      nodes : wayNodes.map((x: INode) => x._id)
+    }});
+}
+
+const dummyData = {
+  nodes: Array<INode>(),

```

Comment:
You can't declare object's properties like this, you have to use either [type alias](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-aliases) or (more preferably) another interface. But since this is supposed to be just a temporary object, I wouldn't declare it at all and just return a generic object with two arrays.

I don't know what does this call do `Array<INode>()`, but you declare arrays in typescript like this `let a: Array<INode>` or (what I prefer more) `let b: INode[]`